deap/tools/selection.py

```diff
@@ -220,107 +235,79 @@ def selLexicase(individuals, k):
         random.shuffle(cases)
 
         while len(cases) > 0 and len(candidates) > 1:
-            f = min
-            if fit_weights[cases[0]] > 0:
-                f = max
-
-            best_val_for_case = f(map(lambda x: x.fitness.values[cases[0]], candidates))
-
-            candidates = list(filter(lambda x: x.fitness.values[cases[0]] == best_val_for_case, candidates))
+            best_val_for_case = max([indiv.fitness.wvalues[cases[0]] for indiv in candidates])
+            candidates = [indiv for indiv in candidates if indiv.fitness.wvalues[cases[0]] == best_val_for_case]
             cases.pop(0)
 
         selected_individuals.append(random.choice(candidates))
 
     return selected_individuals
 
-
-def selEpsilonLexicase(individuals, k, epsilon):
-    """
-    Returns an individual that does the best on the fitness cases when
+def selEpsilonLexicase(individuals, k, epsilon='dynamic'):
+    """Returns an individual that does the best on the fitness cases when
     considered one at a time in random order. Requires a epsilon parameter.
-    https://push-language.hampshire.edu/uploads/default/original/1X/35c30e47ef6323a0a949402914453f277fb1b5b0.pdf
-    Implemented epsilon_y implementation.
 
     :param individuals: A list of individuals to select from.
     :param k: The number of individuals to select.
+    :param epsilon: If an individual is within epsilon of being elite, it will \
+    remain in the selection pool. If 'dynamic', epsilon is set at the start of \
+    each selection even. If 'super-dynamic', epsilon is set realtive to the \
+    current selection pool at each iteration of lexicase selection.
     :returns: A list of selected individuals.
     """
-    selected_individuals = []
-
-    for i in range(k):
-        fit_weights = individuals[0].fitness.weights
 
-        candidates = individuals
-        cases = list(range(len(individuals[0].fitness.values)))
-        random.shuffle(cases)
-
-        while len(cases) > 0 and len(candidates) > 1:
-            if fit_weights[cases[0]] > 0:
-                best_val_for_case = max(map(lambda x: x.fitness.values[cases[0]], candidates))
-                min_val_to_survive_case = best_val_for_case - epsilon
-                candidates = list(filter(lambda x: x.fitness.values[cases[0]] >= min_val_to_survive_case, candidates))
-            else :
-                best_val_for_case = min(map(lambda x: x.fitness.values[cases[0]], candidates))
-                max_val_to_survive_case = best_val_for_case + epsilon
-                candidates = list(filter(lambda x: x.fitness.values[cases[0]] <= max_val_to_survive_case, candidates))
+    # Median Absolute Deviation
+    def _mad(X):
+        median = np.median(X)
+        return np.median([abs(x - median) for x in X])
 
-            cases.pop(0)
-
-        selected_individuals.append(random.choice(candidates))
-
-    return selected_individuals
-
-def selAutomaticEpsilonLexicase(individuals, k):
```

Comment:
Has this function been removed? These kind of changes shouldn't be made since it breaks compatibility with previous 1.X versions.